### PR TITLE
Migrate some more tests away from kIgnoreCertificateErrors

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/19456

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/components/favicon/core/favicon_database_browsertest.cc [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "components/network_session_configurator/common/network_switches.h"
```
Comment (goodov) [inline chromium_src/components/favicon/core/favicon_database_browsertest.cc:14]: can you please move this test from `chromium_src` to `browser`? It doesn't belong here.

File: browser/brave_ads/request_ads_enabled_api_browsertest.cc [hunk 2]
```diff
@@ -93,9 +95,18 @@ class RequestAdsEnabledApiTestBase : public InProcessBrowserTest {
...
+  }
+
+  void SetUpInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::SetUpInProcessBrowserTestFixture();
+    mock_cert_verifier_.SetUpInProcessBrowserTestFixture();
+  }
+
+  void TearDownInProcessBrowserTestFixture() override {
+    InProcessBrowserTest::TearDownInProcessBrowserTestFixture();
+    mock_cert_verifier_.TearDownInProcessBrowserTestFixture();
```
[the remark this answers is not in the document]
Comment (pilgrim-brave) [reply browser/brave_ads/request_ads_enabled_api_browsertest.cc:106]: Fixed in https://github.com/brave/brave-core/commit/8f46b87009d8917934ef7d3f32d148f1f5aeb837

File: browser/brave_ads/request_ads_enabled_api_browsertest.cc [hunk 3]
```diff
@@ -93,9 +95,18 @@ class RequestAdsEnabledApiTestBase : public InProcessBrowserTest {
   }
 
   void SetUpCommandLine(base::CommandLine* command_line) override {
-    // HTTPS server only serves a valid cert for localhost, so this is needed
-    // to load pages from other hosts without an error.
-    command_line->AppendSwitch(switches::kIgnoreCertificateErrors);
```
[the remark this answers is not in the document]
Comment (pilgrim-brave) [reply browser/brave_ads/request_ads_enabled_api_browsertest.cc:98]: Fixed in https://github.com/brave/brave-core/commit/8f46b87009d8917934ef7d3f32d148f1f5aeb837

File: browser/brave_ads/request_ads_enabled_api_browsertest.cc [hunk 4]
```diff
@@ -60,6 +61,7 @@ class RequestAdsEnabledApiTestBase : public InProcessBrowserTest {
 
   void SetUpOnMainThread() override {
     InProcessBrowserTest::SetUpOnMainThread();
+    mock_cert_verifier_.mock_cert_verifier()->set_default_result(net::OK);
     host_resolver()->AddRule("*", "127.0.0.1");
 
     https_server_.reset(new net::EmbeddedTestServer(
```
[the remark this answers is not in the document]
Comment (pilgrim-brave) [reply browser/brave_ads/request_ads_enabled_api_browsertest.cc:66]: Fixed in https://github.com/brave/brave-core/commit/8f46b87009d8917934ef7d3f32d148f1f5aeb837

**Review by iefremov — APPROVED** [review]

**Review by goodov — APPROVED** [review]

**Pull request merged**